libvast/builtins/printers/json.cpp

```diff
@@ -0,0 +1,73 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/arrow_table_slice.hpp>
+#include <vast/concept/printable/vast/json.hpp>
+#include <vast/plugin.hpp>
+
+#include <arrow/record_batch.h>
+
+namespace vast::plugins::json_printer {
+
+class plugin : public virtual printer_plugin {
+public:
+  [[nodiscard]] auto
+  make_printer(const record& options, [[maybe_unused]] type input_schema,
+               const operator_control_plane&) const
+    -> caf::expected<printer_plugin::printer> override {
+    auto input_type = caf::get<record_type>(input_schema);
+    return [input_type, plugin_options = options](
+             generator<table_slice> slices) -> generator<chunk_ptr> {
+      // JSON printer should output NDJSON, see:
+      // https://github.com/ndjson/ndjson-spec
+      auto printer = vast::json_printer{{.oneline = true}};
+      for (const auto& slice : slices) {
+        auto result = std::string{};
```

Comment:
```suggestion
      auto buffer = std::vector<char>{};
      for (const auto& slice : slices) {
```

This has to be a `vector<char>` so you don't have trailing null bytes in the chunk you yield. It can also be outside of the loop as you clear it out with `std::exchange` right before yielding anyways.

Reply:
Ah, yes, the classic debugging `string`. I replaced it with a loop-global `std::vector<char>` in https://github.com/tenzir/vast/pull/3009/commits/71d99569909b08a70f61bfa4f302e3189bf94a2c